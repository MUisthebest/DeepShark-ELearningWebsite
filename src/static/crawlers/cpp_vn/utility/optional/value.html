 <h1 class="firstHeading" id="firstHeading">STD :: Tùy chọn &lt;t&gt; :: Value</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">ConstExpr t &amp; value () &amp;;</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">constexpr const t &amp; value () const &amp;;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx17">(C ++ 20)</span> </td> </tr> </table> <p>Nếu như<code>*this</code>constexpr t &amp;&amp; value () &amp;&amp;;</p>
<p>constexpr const t &amp;&amp; value () const &amp;&amp;;<code><a href="bad_optional_access" title="cpp/utility/optional/bad optional access">std::bad_optional_access</a></code>chứa một giá trị, trả về một tham chiếu đến giá trị chứa.</p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <p>Gán cho đã cho</p>
<h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>Nếu không, ném a</p>
<h3 id="Exceptions">Ngoại lệ</h3> <p><code><a href="bad_optional_access" title="cpp/utility/optional/bad optional access">std::bad_optional_access</a></code>nếu như<code>*this</code>std :: phạm vi :: filter_view :: pre</p>
<h3 id="Notes">Ghi chú</h3> <p>Thành viên đánh máy<code><a href="operator*" title="cpp/utility/optional/operator*">operator*()</a></code>ngoại lệ.<code>value()</code>.</p>
<h3 id="Example">Ví dụ</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">Một tham chiếu đến giá trị chứa.</pre></div> <p>Nếu (sai)</p>
<div class="text source-text"><pre data-language="c">không kiểm tra xem tùy chọn này có chứa giá trị không, có thể hiệu quả hơn</pre></div> </div> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="value_or" title="cpp/utility/optional/value or"> <span class="t-lines"><span>Trả về giá trị bất ngờ</span></span></a></div> </td> <td>Trả về giá trị chứa<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="operator*" title="cpp/utility/optional/operator*"> <span class="t-lines"><span>toán tử-&gt;</span><span>toán tử/=</span></span></a></div> </td> <td>/*di chuyển-box*/<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="bad_optional_access" title="cpp/utility/optional/bad optional access"> <span class="t-lines"><span>bad_optional_access</span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx17">Dest_at</span></span></span></div> </td> <td>Ngoại lệ cho biết đã kiểm tra quyền truy cập vào tùy chọn không chứa giá trị<br/> <span class="t-mark">Các loại chính sách thực thi</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/optional/value">https://en.cppreference.com/w/cpp/utility/optional/value</a>
</p>
</div>
