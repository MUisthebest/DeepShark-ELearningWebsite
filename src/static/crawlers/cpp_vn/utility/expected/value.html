 <h1 class="firstHeading" id="firstHeading">std :: mong đợi &lt;t, e&gt; :: value</h1> <table class="t-dcl-begin"> <tr class="t-dcl-h"> <td> <pre data-language="cpp">Vô hiệu</pre>
</td> <td></td> <td></td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">ConstExpr t &amp; value () &amp;;</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr const t &amp; value () const &amp;;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr t &amp;&amp; value () &amp;&amp;;</pre>
</td> <td>(3)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr const t &amp;&amp; value () const &amp;&amp;;</pre>
</td> <td>(4)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl-h"> <td> <pre data-language="cpp">Vô hiệu</pre>
</td> <td></td> <td></td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">ConstExpr void value () const &amp;;</pre>
</td> <td>             Không có p);</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">ConstExpr void value () &amp;&amp;;</pre>
</td> <td>Consexpr bool any_of (đầu vào đầu tiên, đầu vào cuối cùng, không được tính p);</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Nếu như<code>*this</code>chứa một giá trị dự kiến, trả về một tham chiếu đến giá trị chứa. Trả lại không có gì nếu<code>T</code>Chuyển đổi tiêu chuẩn được áp dụng để chuyển đổi biểu thức thành prvalue.<span class="kw4">Vô hiệu</span>.</p>
<p>Nếu không, ném một ngoại lệ của loại<a href="http://en.cppreference.com/w/cpp/utility/expected/bad_expected_access"><span class="kw3537">Điều đó cho tất cả<span class="me2">bad_expected_access</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">Điều đó cho tất cả<span class="me2">Tìm kiếm phạm vi</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">2,4)</span>chứa một bản sao của<a href="error" title="cpp/utility/expected/error"><code>error()</code></a>.</p>
<div class="t-li1">
<span class="t-li">Đầu vào</span>Nếu như<code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">Điều đó cho tất cả<span class="me2">(c)</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>khoảng cách<code>false</code>, chương trình là xấu.</div> <div class="t-li1">
<span class="t-li">, hành vi là không xác định.</span>Nếu như<code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">Điều đó cho tất cả<span class="me2">(c)</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">Điều đó cho tất cả<span class="me2">, sao chép-khởi tạo phần tử tương ứng của kết quả với</span></span></a><span class="sy1">is_trivently_destructible_v</span>E, decltype<span class="br0">.</span>std<span class="sy4">::</span><span class="me2">(Niebloid)</span><span class="br0">.</span>error<span class="br0">.</span><span class="br0">)</span><span class="br0">)</span><span class="br0">)</span><span class="sy1">&lt;</span></code>khoảng cách<code>false</code>, chương trình là xấu.</div> <div class="t-li1">
<span class="t-li">Mẫu &lt;class inputit, class unyredicate&gt;</span>Nếu như<code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">Điều đó cho tất cả<span class="me2">(c)</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>khoảng cách<code>false</code>, chương trình là xấu.</div> <div class="t-li1">
<span class="t-li">từ_sys</span>Nếu như<code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">Điều đó cho tất cả<span class="me2">hoán đổi void (t2 (&amp; a) [n], t2 (&amp; b) [n]) noExcept ( / * xem bên dưới * /);</span></span></a><span class="sy1">is_trivently_destructible_v</span>E<span class="sy1">&lt;</span></code>khoảng cách<code>false</code>, chương trình là xấu.</div> <h3 id="Parameters">) với giá trị ban đầu</h3> <p>Gán cho đã cho</p>
<h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <div class="t-li1">
<span class="t-li">              Forwardit2 first2,</span>Giá trị dự kiến ​​có trong<code>*this</code>.</div> <div class="t-li1">
<span class="t-li">    trả về std :: find_if_not (đầu tiên, cuối cùng, p) == cuối cùng;</span>Gán cho đã cho</div> <h3 id="Exceptions">Ngoại lệ</h3> <div class="t-li1">
<span class="t-li">1,2,5)</span>Ném<code><a href="http://en.cppreference.com/w/cpp/utility/expected/bad_expected_access"><span class="kw3537">Điều đó cho tất cả<span class="me2">bad_expected_access</span></span></a><span class="br0">.</span><a href="http://en.cppreference.com/w/cpp/utility/as_const"><span class="kw1124">Điều đó cho tất cả<span class="me2">as_const</span></span></a><span class="br0">.</span>error<span class="br0">.</span><span class="br0">)</span><span class="br0">)</span><span class="br0">)</span></code>nếu như<code>*this</code>chứa một giá trị bất ngờ.</div> <div class="t-li1">
<span class="t-li">3,4,6)</span>Ném<code><a href="http://en.cppreference.com/w/cpp/utility/expected/bad_expected_access"><span class="kw3537">Điều đó cho tất cả<span class="me2">bad_expected_access</span></span></a><span class="br0">.</span>std<span class="sy4">::</span><span class="me2">(Niebloid)</span><span class="br0">.</span>error<span class="br0">.</span><span class="br0">)</span><span class="br0">)</span><span class="br0">)</span></code>nếu như<code>*this</code>chứa một giá trị bất ngờ.</div> <h3 id="Example">Ví dụ</h3> <h3 id="Defect_reports">Báo cáo khiếm khuyết</h3> <p>Các báo cáo khiếm khuyết thay đổi hành vi sau đây được áp dụng hồi tố cho các tiêu chuẩn C ++ được công bố trước đó.</p>
<table class="dsctable"> <tr> <th>Dr</th> <th>Áp dụng cho</th> <th>Hành vi như được công bố</th> <th>Hành vi chính xác</th>
</tr> <tr> <td>
<a class="external text" href="https://cplusplus.github.io/LWG/issue3940" rel="nofollow">LWG 3940</a> </td> <td>C ++ 23</td> <td>
<code>E</code>không bắt buộc phải được sao chép/di chuyển để chế tạo cho quá tải<span class="t-v">Mẫu &lt;Class execPolicy, class forwardit1, class forwardit2,</span> </td> <td>Thực hiện quá tải</td>
</tr>
</table> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="value_or" title="cpp/utility/expected/value or"> <span class="t-lines"><span>Trả về giá trị bất ngờ</span></span></a></div> </td> <td>value_or<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="operator*" title="cpp/utility/expected/operator*"> <span class="t-lines"><span>toán tử-&gt;</span><span>toán tử/=</span></span></a></div> </td> <td>đối tượng, cùng với giá trị chứa của nó<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="error" title="cpp/utility/expected/error"> <span class="t-lines"><span>. Nếu thất bại đã được gây ra bởi một số lỗi khác, hãy đặt</span></span></a></div> </td> <td>Trả về giá trị dự kiến<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="bad_expected_access" title="cpp/utility/expected/bad expected access"> <span class="t-lines"><span>bad_expected_access</span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx23">Phạm vi :: IOTA</span></span></span></div> </td> <td>Ngoại lệ cho biết quyền truy cập đã kiểm tra vào một<code>expected</code>chứa một giá trị bất ngờ<br/> <span class="t-mark">Kiểm tra xem một lớp có đại diện cho chính sách thực thi không</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/expected/value">https://en.cppreference.com/w/cpp/utility/expected/value</a>
</p>
</div>
